gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/InstantiatingHttpJsonChannelProvider.java

```diff
@@ -119,11 +119,33 @@ public boolean needsEndpoint() {
     return endpoint == null;
   }
 
+  @Override
+  public boolean needsMtlsEndpoint() {
+    return false;
+  }
+
+  @Override
+  public boolean needsEndpointOverride() {
+    return false;
+  }
+
   @Override
   public TransportChannelProvider withEndpoint(String endpoint) {
     return toBuilder().setEndpoint(endpoint).build();
   }
 
+  @Override
+  public TransportChannelProvider withMtlsEndpoint(String mtlsEndpoint) {
+    throw new UnsupportedOperationException(
+        "InstantiatingHttpJsonChannelProvider doesn't need an mtlsEndpoint");
+  }
```

Comment:
qq, do you know if mTLS supported for HttpJson? Or is this specific to gRPC?

Reply:
We are only adding mTLS via S2A support to gRPC in Java. This is because our [S2A client](https://github.com/grpc/grpc-java/tree/master/s2a) does not support HTTP (we would like to do this, but currently there is no timeline).

In contrast, in Golang, mTLS via S2A is supported for HTTP and gRPC: https://github.com/googleapis/google-api-go-client/pull/1874